lib/email_processor.rb

```diff
@@ -38,6 +54,10 @@ def define_sender
 
   def define_course
     @course = @sender.courses.last if @sender
+    if @course.nil?
+      slug = retrieve_course_slug_by_url
+      @course = Course.find_by(slug: slug) if slug
```

Comment:
Course slug will never be nil, so you could just do `@course = Course.find_by(slug: retrieve_course_slug_by_url)`.

This is probabably a more reliable matcher than sender's last course, too. Sometimes an instructor has several simultaneous courses.

I suggest:

```ruby
@course = Course.find_by(slug: retrieve_course_slug_by_url)
@course ||= @sender.courses.last if @sender
```